functions/default.pp

```diff
@@ -8,8 +8,7 @@ function postgresql::default(
 ) {
   include postgresql::params
 
-  #search for the variable name in params first
-  #then fall back to globals if not found
-  pick( getvar("postgresql::params::${parameter_name}"),
-  "postgresql::globals::${parameter_name}")
+  # Search for the variable name in params.
+  # params inherits from globals, so it will also catch these variables.
+  pick(getvar("postgresql::params::${parameter_name}"))
```

Comment:
is there a situation where a variable is defined in globales but not params? with your change we wouldn't find a value anymore.

Reply:
params inherits from globals. `postgresql::globals::default_connect_settings` value can be read with `$postgresql::params::default_connect_settings` as shown by the unit test ;-)

Reply:
ah right :D